tests/robustness/traffic/kubernetes.go

```diff
@@ -62,16 +61,18 @@ func (t kubernetesTraffic) Run(ctx context.Context, c *RecordingClient, limiter
 	kc := &kubernetesClient{client: c}
 	s := newStorage()
 	keyPrefix := "/registry/" + t.resource + "/"
-	g := errgroup.Group{}
+	wg := sync.WaitGroup{}
 	readLimit := t.averageKeyCount
 
-	g.Go(func() error {
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
 		for {
 			select {
 			case <-ctx.Done():
-				return ctx.Err()
+				return
```

Comment:
Not returning an error here since we check for it anyway higher up in the call stack: 
https://github.com/etcd-io/etcd/blob/94b155a1c922c301119eb1b3d3e950355a5225d7/tests/robustness/traffic/traffic.go#L101

Also, this way it is more consistent with how the etcdTraffic implementation handles this.